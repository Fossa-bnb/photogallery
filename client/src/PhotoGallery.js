import React from 'react';
import $ from 'jquery';


class PhotoGallery extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      photos: [],
    }
  }

  componentDidMount() {
    //initiate ajax request to the server with a particular id
    //setup a route on the server to receive get request to endpoint /id
    const id = 5;
    $.ajax({
      url: `/${id}`,
      method: "GET",
      success: ({photos}) => {
        console.log(`Successful GET request for photos of ${id}`);
        console.log(`Got back these photos on the client: ${photos.length}`);
        this.setState({
          photos: photos
        });
      },
      error: (err) => {
        console.log(`Error in GET request from client: ${err}`);
      }
    })
  }

  render() {
    return (
      <div className="photo-gallery">
        Photo Gallery Module
      </div>
    )
  }
}

export default PhotoGallery;