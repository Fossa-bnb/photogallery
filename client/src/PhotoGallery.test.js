import React from 'react';
import PhotoGallery from './PhotoGallery';
import { shallow } from 'enzyme';

describe('Test for <PhotoGallery /> Component', () => {
  it('should render a div with class "photo-gallery"', () => {
    const wrapper = shallow(<PhotoGallery />);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.is('div')).toEqual(true);
  });
});