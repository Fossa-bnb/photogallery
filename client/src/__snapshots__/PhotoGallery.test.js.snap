// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <PhotoGallery /> Component should render a div with class "photo-gallery" 1`] = `
<div
  className="photo-gallery"
>
  Photo Gallery Module
</div>
`;
